core/api/test/integration/app/user/consent-list.spec.ts

```diff
@@ -0,0 +1,102 @@
+import { exec } from "child_process"
+import puppeteer from "puppeteer"
+
+import { Admin } from "@/app"
+import { consentList } from "@/services/hydra"
+import { sleep } from "@/utils"
+
+let userId: UserId
+const email = "test@galoy.io" as EmailAddress
+
+beforeAll(async () => {
+  const account = await Admin.getAccountByUserEmail(email)
+  if (account instanceof Error) throw account
+
+  userId = account.kratosUserId
+})
+
+const getOTP = (email: string): Promise<string> => {
+  return new Promise((resolve, reject) => {
+    const query = `docker exec -i galoy-dev-kratos-pg-1 psql -U dbuser -d default -t -c "SELECT body FROM courier_messages WHERE recipient='${email}' ORDER BY created_at DESC LIMIT 1;"`
+    exec(query, (error, stdout, stderr) => {
+      if (error) {
+        reject(`error: ${error.message}`)
+        return
+      }
+      if (stderr) {
+        reject(`stderr: ${stderr}`)
+        return
+      }
+      const otpMatch = stdout.match(/(\d{6})/)
+      if (otpMatch && otpMatch[1]) {
+        resolve(otpMatch[1])
+      } else {
+        reject("OTP not found in the message")
+      }
+    })
+  })
+}
+
+async function performOAuthLogin() {
+  const screenshots = false
+
+  const browser = await puppeteer.launch()
+  // const browser = await puppeteer.launch({ headless: true })
+  const page = await browser.newPage()
+
+  // Navigate the page to a URL
+  await page.goto("http://localhost:3001/api/auth/signin")
+
+  screenshots && (await page.screenshot({ path: "screenshot0.png" }))
```

Comment:
is this only help for debug? what is the purpose of take screenshots?

Reply:
yup indeed to help debugging going over the different screens